app/models/concerns/omniauthable.rb

```diff
@@ -88,5 +86,12 @@ def ensure_unique_username(starting_username)
 
       username
     end
+
+    def ensure_valid_username(starting_username)
+      starting_username = starting_username.split('@')[0]
+      temp_username = starting_username.match(/\A[a-z0-9_]+\z/i) ? starting_username : starting_username.gsub(/[^a-z0-9_]+/i, '')
```

Comment:
No need to conditionally do this, the `gsub` call can be done every time.

However, silently removing characters may lead in unintended collisions, so I'm not sure it should be done at all, or it should be properly documented.

Reply:
> However, silently removing characters may lead in unintended collisions, so I'm not sure it should be done at all, or it should be properly documented.

After this step, we check for uniqueness which should handle any potential collisions. See `ensure_unique_username`.